### ISSUE TITLE
Connection lost before handshake

### ISSUE THREAD

**Issue by shanon00** [electerm/electerm#3538]
### Electerm Version and download file extension(Electerm版本和下载文件后缀)

electerm-1.39.88-mac-arm64.dmg

### Platform detail (平台详情)

ARM 64

### What steps will reproduce the bug?(重新问题的详细步骤)

Connection lost before handshake, When I use proxy, it will report an error.
1. Check Network Connection
2. Firewall Settings
3. Server Status
4. Protocol Compatibility
5. Try a Different Client
I have checked the above content. they are good

### What should have happened?(期望的结果)

can u tell me ，Why can't I connect after connecting to the proxy? and  How to deal with

**Comment by zxdong262** [comment]
In electerm settings, there is a proxy setting, could use that, I got report from other user that use mac os system proxy would cause app crash

**Comment by shanon00** [comment]
Yes, I tried it, but it didn't work. After adding a proxy, it will prompt 'handshake failure'. There is no problem without adding an agent. It may be a problem with my vpn return.

**Comment by Jinming6** [comment]
The VPN is configured as a global proxy and no rule-based proxy is used. It worked for me.

**Comment by duanbingu** [comment]
So did I

**Comment by duanbingu** [comment]
vpn configuration of global proxy or rule-based will encounter handshake failure, and the global proxy is set in the electerm opening

**Comment by badboycxcc** [comment]
MAC 不管是否使用代理都无法连接，同样的错误。
